design.md

```diff
@@ -1,122 +1,210 @@
-```    
-print (4 & 5)
-```
-> "Parse Error: Could not parse operator at line 1"
-
-4.  Parse Error: Invalid syntax
-```
-x: int = 4
-	
-if (x == 4)::
-	print(x)
-```
-> "Parse Error: Could not parse stmt at line 3"
-
-5. Parse Error: Missing type annotation in function definition
-```
-def f(c):
-	print(3)
-
-f(2)
-```
-> "Parse Error: Missing type annotation for parameter c at line 1"
-
-6.  Type Error: Type mismatch
-```
-x: int = 1
-x = True
-```
-> "Type Error: bool type cannot be assigned to int type at line 2"  
-
-7.  Type Error: Function call type mismatch
-```    
-def f(c: int):
-	print(c)
-
-f(2,3)
-```
-  > "Type Error: Function call type mismatch: f at line 1"
-
-8.  Type Error: Variable not defined before accessing
-```    
-z = 1
-```
-> "Type Error: unbound id: z at line 1"
-
-9.  Type Error: Function not defined before calling
-```
-def sum(a: int, b: int) -> int:
-	return a + b
-
-diff(5,7)
-```
-> "Type Error: Undefined function: diff at line 3"  
-
-10. Type Error: Return type mismatch in a function definition
-```
-def f(c: int) -> int:
-	return True
-
-f(2)
-```
-> "Type Error: Expected return type int, got type bool at line 2"  
-
-11. Runtime Error: Operation on none
-```
-class C(object):
-	x: int = 0
-
-c: C = None
-c.x
-```
-> "Runtime Error: Cannot perform operation on none at line 5"
-
-12. Runtime Error: Division by zero
-```
-x: int = 100//0
-```
-> "Runtime Error: Cannot divide by zero at line 1"
\ No newline at end of file
+Jeff Makings and Zach Lee 
+
+Test cases: 
+1.
+Test for initial compilation: 
+“999999999999999999999999999999999999999999999999” 
+Expected: 999999999999999999999999999999999999999999999999
+2.
+Basic assignment and printing: 
+“x: int = 999999999999999999999999999999999999999999999999 
+print(x)” 
+Expected: “999999999999999999999999999999999999999999999999” 
+3.
+Unary operation: 
+“x: int = -999999999999999999999999999999999999999999999999 
+print(x)” 
+Expected: “-999999999999999999999999999999999999999999999999”
+4.
+Basic addition: 
+“x:int = 1000000000000000000000000000 + 1000000000000000000000000000
+print(x)” 
+Expected: 2000000000000000000000000000
+5.
+Basic subtraction: 
+“x:int = 5000000000000000000000000000 - 2000000000000000000000000000
+print(x)” 
+Expected: 3000000000000000000000000000
+6.
+Multiply i32s to get BigNum: 
+“x:int = 10000*300000
+print(x)” 
+Expected: 3000000000
+7.
+Divide: 
+“x:int = 30000000000000000000000000000 / 30000000000 
+print(x)”
+Expected: 1000000000000000000
+8.
+Comparison Operator: 
+“x:int = 1000000000000000000
+y: int = 2000000000000000000
+If y > x: 
+	print(True)
+Else: 
+	print(False)
+Expected: True” 
+9.
+Equality Operator: 
+“x:int = 1000000000000000000
+y: int = 2000000000000000000
+If y == x: 
+	print(True)
+Else: 
+	print(False)
+Expected: False” 
+10.
+List Test (for when it’s implemented): 
+“x:int = 900000001
+list1 = [3] * x
+print(list1[x-1])”
+Expected: 3
+
+The below is related to the concerns from instructors.
+
+(1)
```

Comment:
Thanks for explaining all the changes in each file!

Reply:
Thanks Vignesh for your constructive comments, we'll try to implement all the changes you suggested in the coming days. We'll let you know if we have any questions/clarifications.